source/client/options_impl.cc

```diff
@@ -240,6 +252,16 @@ OptionsImpl::OptionsImpl(int argc, const char* const* argv) {
                        upper_cased, &sequencer_idle_strategy_),
                    "Failed to parse sequencer idle strategy");
   }
+
+  if (h1_connection_reuse_strategy.isSet()) {
+    std::string upper_cased = h1_connection_reuse_strategy.getValue();
+    absl::AsciiStrToUpper(&upper_cased);
+    RELEASE_ASSERT(
```

Comment:
Should these be exceptions and we catch them at the top? I think this makes testing easier.

Reply:
This should be invariant; we have a problem when we would raise that exception as that means we didn't set up TCLAP's constraints at all. In the regular user-flow, entering a value not listed as valid will result in TCLAP throwing an exception, which we will handle and show as an error to the user.

Added some testing around this in 49e4d2dfbf3e3c3a0a7a9d7bd5befab0c1192c5e

Reply:
If it's an invariant then this would ideally be ASSERT.